core/os/android/adb/commands.go

```diff
@@ -496,29 +496,50 @@ func (b *binding) DriverVersionCode(ctx context.Context) (int, error) {
 	return ip.VersionCode, err
 }
 
-// HasGpuProfilingSupportInSystemImage returns whether system image has GPU profiling support.
-func (b *binding) HasGpuProfilingSupportInSystemImage(ctx context.Context) (bool, error) {
-	supported, err := b.SystemProperty(ctx, systemImageGpuProfilerSupportProperty)
+// PrepareGpuProfiling queries GPU profiling support, and when profiling is supported it sets up
+// the device for profiling of installedPackage app. It returns:
+// - a bool which is true when GPU profiling is supported and the setup is done without errors
+// - a string that contains the name of the package where to find profiling layers, if profiling
+//   is supported by the system image, then this string is empty
+// - a cleanup function to revert the device settings after profiling is done
+// - an error to indicate if anything went wrong
+// The returned bool disambiguates between "an error happened" and "profiling is not supported".
```

Comment:
This documentation should be on the interface not here, I think.

Reply:
done